crates/server/src/datastore/memory.rs

```diff
@@ -577,4 +578,27 @@ impl DataStore for MemoryDataStore {
             checkpoint,
         })
     }
+
+    async fn verify_package_record_signature(
```

Comment:
What is this function doing?

Reply:
We need to verify the signature of the record being published prior to accepting the record for processing; to do that, the data store needs to fetch the package's validation state to determine the public key to verify with.

This prevents people from sending us requests where the `keyId` property is explicitly set to an acceptable (known) key id, as these are not private, and then giving us a record signed with a mismatched private key.

By doing so, it preemptively rejects the record before it ever makes it into the database for later validation; validation would have rejected the record anyway due to the invalid signature.

As authorization is based on these public key ids, we shouldn't allow unauthorized keys to write to the database in any form.